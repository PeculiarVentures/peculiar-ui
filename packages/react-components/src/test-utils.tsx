import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
