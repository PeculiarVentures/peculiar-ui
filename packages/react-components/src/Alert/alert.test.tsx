import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Alert } from './index';

describe('<Alert />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <Alert>
        Alert message
      </Alert>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <Alert className="my-class-name">
        Alert message
      </Alert>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('variants', () => {
    const variants: Array<React.ComponentProps<typeof Alert>['variant']> = [
      'attention',
      'pending',
      'success',
      'wrong',
    ];

    variants.forEach((variant) => {
      describe(`variant "${variant}"`, () => {
        it('enable icon', () => {
          const { asFragment } = render(
            <Alert variant={variant} disableIcon={false}>
              Message
              {variant}
            </Alert>,
          );

          expect(asFragment()).toMatchSnapshot();
        });

        it('disable icon', () => {
          const { asFragment } = render(
            <Alert variant={variant} disableIcon>
              Message
              {variant}
            </Alert>,
          );

          expect(asFragment()).toMatchSnapshot();
        });
      });
    });
  });
});
