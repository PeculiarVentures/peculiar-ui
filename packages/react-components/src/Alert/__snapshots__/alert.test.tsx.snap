// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="my-class-name emotion-0"
    role="alert"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Alert message
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Alert message
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "attention" disable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="attention"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Messageattention
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "attention" enable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--pv-color-attention);
}

.emotion-2 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-3 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="attention"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.57 4.385c1.066-1.847 3.73-1.847 4.796 0l7.195 12.461C22.627 18.692 21.294 21 19.163 21H4.773c-2.131 0-3.464-2.308-2.398-4.154L9.57 4.385z"
          fill="currentColor"
        />
        <circle
          cx="11.968"
          cy="16.846"
          fill="#fff"
          r="1.385"
        />
        <path
          d="M10.227 11.11l1.741 4.351 1.74-4.351a1.875 1.875 0 10-3.48 0z"
          fill="#fff"
        />
      </svg>
    </div>
    <div
      class="emotion-2"
    >
      <p
        class="emotion-3"
      >
        Messageattention
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "pending" disable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="pending"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Messagepending
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "pending" enable icon 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  overflow: hidden;
  position: relative;
  display: inline-block;
  color: var(--pv-color-secondary);
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
  height: var(--pv-size-base-3);
  width: var(--pv-size-base-3);
}

.emotion-3 {
  display: block;
}

.emotion-4 {
  stroke: currentcolor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

.emotion-5 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-6 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="pending"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-hidden="true"
        class="emotion-2"
        role="progressbar"
      >
        <svg
          class="emotion-3"
          viewBox="22 22 44 44"
        >
          <circle
            class="emotion-4"
            cx="44"
            cy="44"
            fill="none"
            r="20"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <p
        class="emotion-6"
      >
        Messagepending
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "success" disable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="success"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Messagesuccess
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "success" enable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--pv-color-success);
}

.emotion-2 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-3 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="success"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="currentColor"
          height="21.333"
          rx="10.667"
          width="21.333"
          x="1.333"
          y="1.333"
        />
        <path
          d="M8 10.667L11.333 14l5-5"
          stroke="#fff"
          stroke-width="1.5"
        />
      </svg>
    </div>
    <div
      class="emotion-2"
    >
      <p
        class="emotion-3"
      >
        Messagesuccess
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "wrong" disable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-wrong-tint-5);
}

.emotion-1 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="wrong"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-2"
      >
        Messagewrong
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> variants variant "wrong" enable icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  gap: var(--pv-size-base-2);
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-wrong-tint-5);
}

.emotion-1 {
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--pv-color-wrong);
}

.emotion-2 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: var(--pv-size-base) 0px;
  min-height: var(--pv-text-b3-height);
  overflow-wrap: break-word;
}

.emotion-3 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<div
    class="emotion-0"
    role="alert"
    variant="wrong"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="currentColor"
          height="21.333"
          rx="10.667"
          width="21.333"
          x="1"
          y="1"
        />
        <path
          clip-rule="evenodd"
          d="M12.185 11.5l2.721 2.72a.324.324 0 010 .457l-.23.229a.323.323 0 01-.456 0l-2.72-2.72-2.72 2.72a.324.324 0 01-.457 0l-.229-.23a.324.324 0 010-.456l2.721-2.72-2.721-2.72a.324.324 0 010-.457l.229-.229a.324.324 0 01.457 0l2.72 2.72 2.72-2.72a.323.323 0 01.457 0l.229.23a.324.324 0 010 .456l-2.721 2.72z"
          fill="#fff"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="emotion-2"
    >
      <p
        class="emotion-3"
      >
        Messagewrong
      </p>
    </div>
  </div>
</DocumentFragment>
`;
